src/Microsoft.Extensions.Configuration.AzureAppConfiguration/FeatureManagement/FeatureManagementKeyValueAdapter.cs

```diff
@@ -247,6 +240,995 @@ public bool NeedsRefresh()
+
+                        break;
+                }
+            }
+
+            return featureVariant;
+        }
+
+        private FeatureTelemetry ParseFeatureTelemetry(ref Utf8JsonReader reader, string settingKey)
+        {
+            var featureTelemetry = new FeatureTelemetry();
+
+            while (reader.Read() && reader.TokenType != JsonTokenType.EndObject)
+            {
+                if (reader.TokenType != JsonTokenType.PropertyName)
+                {
+                    continue;
+                }
+
+                string telemetryPropertyName = reader.GetString();
+
+                switch (telemetryPropertyName)
+                {
+                    case FeatureManagementConstants.Enabled:
+                        {
+                            if (reader.Read() && (reader.TokenType == JsonTokenType.False || reader.TokenType == JsonTokenType.True))
+                            {
+                                featureTelemetry.Enabled = reader.GetBoolean();
+                            }
+                            else if (reader.TokenType == JsonTokenType.String && bool.TryParse(reader.GetString(), out bool enabled))
+                            {
+                                featureTelemetry.Enabled = enabled;
+                            }
+                            else if (reader.TokenType != JsonTokenType.Null)
+                            {
+                                throw CreateFeatureFlagFormatException(
+                                    FeatureManagementConstants.Enabled,
+                                    settingKey,
+                                    reader.TokenType.ToString(),
+                                    $"{JsonTokenType.True}' or '{JsonTokenType.False}");
+                            }
+
+                            break;
+                        }
+
+                    case FeatureManagementConstants.Metadata:
+                        {
+                            if (reader.Read() && reader.TokenType == JsonTokenType.StartObject)
+                            {
+                                featureTelemetry.Metadata = new Dictionary<string, string>();
+
+                                while (reader.Read() && reader.TokenType != JsonTokenType.EndObject)
+                                {
+                                    if (reader.TokenType != JsonTokenType.PropertyName)
+                                    {
+                                        continue;
+                                    }
+
+                                    string metadataPropertyName = reader.GetString();
+
+                                    if (reader.Read() && reader.TokenType == JsonTokenType.String)
+                                    {
+                                        featureTelemetry.Metadata[metadataPropertyName] = reader.GetString();
+                                    }
+                                    else if (reader.TokenType != JsonTokenType.Null)
```

Comment:
So null is not a valid value here? We might not need the null check and throw for anything thats not a string.

Reply:
Oh `null` is a valid string, since `JsonSerializer.Deserialize` would treat null as a valid string. In other cases for strings it wouldn't matter adding them since referencing configuration would treat it the same way, getting a missing key would just return null. For that reason, I thought it would be the same case here because adding the key or not results in the same behavior when reading the configuration. Let me know if I missed something there though